src/transforms/remap.rs

```diff
@@ -133,14 +135,50 @@ pub struct RemapConfig {
     #[configurable(derived, metadata(docs::hidden))]
     #[serde(default)]
     pub runtime: VrlRuntime,
+
+    #[configurable(derived, metadata(docs::hidden))]
+    #[serde(skip)]
+    #[derivative(Debug = "ignore")]
+    pub cache: Mutex<
+        Vec<(
+            (TableRegistry, schema::Definition),
+            std::result::Result<(Program, String, MeaningList), String>,
+        )>,
+    >,
+}
+
+impl Clone for RemapConfig {
+    fn clone(&self) -> Self {
+        Self {
+            source: self.source.clone(),
+            file: self.file.clone(),
+            metric_tag_values: self.metric_tag_values,
+            timezone: self.timezone,
+            drop_on_error: self.drop_on_error,
+            drop_on_abort: self.drop_on_abort,
+            reroute_dropped: self.reroute_dropped,
+            runtime: self.runtime,
+            cache: Mutex::new(Default::default()),
+        }
+    }
 }
 
 impl RemapConfig {
     fn compile_vrl_program(
         &self,
-        enrichment_tables: vector_lib::enrichment::TableRegistry,
+        enrichment_tables: TableRegistry,
         merged_schema_definition: schema::Definition,
-    ) -> Result<(Program, String, Vec<Box<dyn Function>>, CompileConfig)> {
+    ) -> Result<(Program, String, MeaningList)> {
+        if let Some((_, res)) = self
+            .cache
+            .lock()
+            .unwrap()
```

Comment:
Ditto use `expect` here too, as above.

Also, it looks like the pattern of `self.cache.lock().expect()`  could be put into a helper function to hide that complexity.

Reply:
I don't think helper function is worth it. And honestly I can't think for a good name for it) 
`lock_unpoisoned(s: &str)` too long
`lockp`/`lock_p`/`lock_up` doesn't tell what it does